### PR TITLE
Kl 2021 08 check permissions in app serializer

### PR DESCRIPTION
**Pull request by fuzzylogic2000**

### CODE REVIEW

**Comment by fuzzylogic2000** [comment]
Just a second, a few more lines to the tests and the missing lines are also covered!

File: apps/ideas/serializers.py [hunk 1]
```diff
@@ -86,6 +91,30 @@ def get_user_rating(self, idea):
                     return RatingSerializer(ratings.first()).data
         return None
 
+    def get_has_rating_permission(self, idea):
+        user = None
+        request = self.context.get('request')
+        if request and hasattr(request, 'user'):
+            user = request.user
+            return user.has_perm('a4_candy_ideas.rate_idea', idea)
+        return False
```
Comment (fuzzylogic2000) [inline apps/ideas/serializers.py:100]: > Just a second, a few more lines to the tests and the missing lines are also covered!

Well, even when we have no user logged in, the API gets the anonymous user, so I can't test these lines. But I would still like to keep them as is. So, test coverage is falling. :woman_shrugging:

**Comment by fuzzylogic2000** [comment]
Alright, done! And can be checked together with https://github.com/liqd/zt-app/pull/85
@Rineee @phillimorland @goapunk

File: apps/ideas/serializers.py [hunk 2]
```diff
@@ -86,6 +91,30 @@ def get_user_rating(self, idea):
                     return RatingSerializer(ratings.first()).data
         return None
 
+    def get_has_rating_permission(self, idea):
+        user = None
```
Comment (Rineee) [inline apps/ideas/serializers.py:95]: I think this is not needed, right?

Comment (fuzzylogic2000) [reply apps/ideas/serializers.py:95]: Yes, you're right! Will remove it.

**Pull request merged**